concordium-consensus/src/Concordium/GlobalState/Account.hs

```diff
@@ -230,8 +229,10 @@ data EncryptedAmountUpdate =
 
 -- |An update to an account state.
 data AccountUpdate = AccountUpdate {
-  -- |Address of the affected account.
-  _auAddress :: !AccountAddress
+  -- |Index of the affected account.
+  _auIndex :: !AccountIndex
+  -- |Canonical address of the affected account.
+  ,_auAddress :: !AccountAddress
```

Comment:
The key in that map is now an AccountIndex, not an account address. We could in principle look up the account address from the index there, but I am hesitant to do that since it would introduce extra work and might significantly affect performance which we do not have time to adequately test at the moment.

Reply:
I also don't like the fact there are two, but it is something that can be revised later.

Reply:
Then `bsoModifyAccount` can be made to return the canonical address for use in `commitChanges`.